tests/math/math.cpp

```diff
@@ -0,0 +1,132 @@
+/* RUN: %{execute}%s
+
+   Test the behaviour of math.cpp
+*/
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <boost/test/minimal.hpp>
+
+using namespace cl::sycl;
+
+#define TRYSYCL_MATH_WRAPX(FUN, i) std::cout << FUN(i) << std::endl;
+#define TRYSYCL_MATH_WRAPX2(FUN, i) std::cout << FUN(i, i) << std::endl;
+#define TRYSYCL_MATH_WRAPX2s(FUN, i) std::cout << FUN(i, i) << std::endl;
+#define TRYSYCL_MATH_WRAPX3(FUN, i) std::cout << FUN(i, i, i) << std::endl;
+#define TRYSYCL_MATH_WRAPX3s(FUN, i) std::cout << FUN(i, i, i) << std::endl;
+#define TRYSYCL_MATH_WRAPX3ss(FUN, i) std::cout << FUN(i, i, i) << std::endl;
+
+int test_main(int argc, char *argv[]) {
+  constexpr auto N = 100;
+  for(int k = 0; k < N; k++)
```

Comment:
If you have auto above, why not pushing the laziness and having auto instead of int too?